tests/admin_inlines/admin.py

```diff
@@ -342,6 +343,59 @@ class ClassAdminStackedVertical(admin.ModelAdmin):
     inlines = [ClassStackedVertical]
 
 
+# admin and form for #31867
+class Child1Form(forms.ModelForm):
+
+    class Meta:
+        fields = '__all__'
+        model = Child1
+        widgets = {
+            'position': forms.HiddenInput,
+        }
+
+    def _post_clean(self):
+        super()._post_clean()
+        if self.instance is not None and self.instance.position == 1:
+            self.add_error(None, ValidationError("A non-field error"))
+
+
+class Child1Inline(admin.TabularInline):
+    model = Child1
+    form = Child1Form
+
+
+class Child2Form(forms.ModelForm):
+
+    class Meta:
+        fields = '__all__'
+        model = Child2
+        widgets = {
+            'position': forms.HiddenInput,
+        }
+
+
+class Child2Inline(admin.StackedInline):
+    model = Child2
+    form = Child2Form
+    fields = (("name", "position"), )
```

Comment:
Use list and remove unnecessary whitespace
```suggestion
    fields = [('name', 'position')]
```